flink-connector/src/test/java/com/datastrato/gravitino/flink/connector/integration/test/hive/HiveCatalogStoreIT.java

```diff
@@ -0,0 +1,236 @@
+
+    // Drop the catalog. Only support drop catalog by SQL.
+    tableEnv.useCatalog(DEFAULT_CATALOG);
+    tableEnv.executeSql("drop catalog " + catalogName);
+    Assertions.assertFalse(
+        metalake.catalogExists(NameIdentifier.ofCatalog(FlinkEnvIT.metalakeName, catalogName)));
+
+    Optional<Catalog> droppedCatalog = tableEnv.getCatalog(catalogName);
+    Assertions.assertFalse(droppedCatalog.isPresent(), "Catalog should be dropped");
+  }
+
+  @Test
+  public void testCreateGravitinoHiveCatalogUsingSQL() {
+    tableEnv.useCatalog(DEFAULT_CATALOG);
+
+    // Create a new catalog.
+    String catalogName = "gravitino_hive_sql";
+    Configuration configuration = new Configuration();
+    configuration.set(CommonCatalogOptions.CATALOG_TYPE, GravitinoHiveCatalogFactory.IDENTIFIER);
+    configuration.set(HiveCatalogFactoryOptions.HIVE_CONF_DIR, "src/test/resources/flink-tests");
+    tableEnv.executeSql(
+        String.format(
+            "create catalog %s with ('type'='gravitino-hive', "
+                + "'hive-conf-dir'='src/test/resources/flink-tests')",
+            catalogName));
+    Assertions.assertTrue(
+        metalake.catalogExists(NameIdentifier.ofCatalog(metalakeName, catalogName)));
+
+    // Get the created catalog.
+    Optional<org.apache.flink.table.catalog.Catalog> catalog = tableEnv.getCatalog(catalogName);
+    Assertions.assertTrue(catalog.isPresent());
+    Assertions.assertInstanceOf(GravitinoHiveCatalog.class, catalog.get());
+
+    // List catalogs.
+    String[] catalogs = tableEnv.listCatalogs();
+    Assertions.assertEquals(2, catalogs.length, "Should create a new catalog");
+    Assertions.assertTrue(
+        Arrays.asList(catalogs).contains(catalogName), "Should create the correct catalog.");
+
+    // Use SQL to list catalogs.
+    TableResult result = tableEnv.executeSql("show catalogs");
+    Assertions.assertEquals(
+        2, Lists.newArrayList(result.collect()).size(), "Should have 2 catalogs");
+
+    Assertions.assertEquals(
+        DEFAULT_CATALOG,
+        tableEnv.getCurrentCatalog(),
+        "Current catalog should be default_catalog in flink");
+
+    // Change the current catalog to the new created catalog.
+    tableEnv.useCatalog(catalogName);
+    Assertions.assertEquals(
+        catalogName,
+        tableEnv.getCurrentCatalog(),
+        "Current catalog should be the one that is created just now.");
+
+    // Drop the catalog. Only support using SQL to drop catalog.
+    tableEnv.useCatalog(DEFAULT_CATALOG);
+    tableEnv.executeSql("drop catalog " + catalogName);
+    Assertions.assertFalse(
+        metalake.catalogExists(NameIdentifier.ofCatalog(FlinkEnvIT.metalakeName, catalogName)));
+
+    Optional<Catalog> droppedCatalog = tableEnv.getCatalog(catalogName);
+    Assertions.assertFalse(droppedCatalog.isPresent(), "Catalog should be dropped");
+  }
+
+  @Test
+  public void testCreateHiveCatalogInMemoryStore() {
```

Comment:
why add the test for memory store?

Reply:
@FANNG1 I think it's not necessary.  It just gives an example to show how to register the GravitinoHiveCatalog when using InMemoryStore. The original idea was to how to use it below Flink1.18. But I think it is not as expected. I will delete them. I will add them back when we support multiple Flink versions.